src/core/tensor/tensor.cc

```diff
@@ -685,8 +685,7 @@ void Axpy(const SType alpha, const Tensor &in, Tensor *out) {
     }, {in.blob(), out->blob()}, {out->blob()});
   });
 }
-template <>
-void Axpy(const float alpha, const Tensor &in, Tensor *out);
+template void Axpy(const float alpha, const Tensor &in, Tensor *out);
```

Comment:
The explicit instantiation is because the definition of Axpy() function is seperated from its declaration.
We'd better move the definition of template functions into tensor.h
Ref: https://isocpp.org/wiki/faq/templates#templates-defn-vs-decl